src/components/settings/PushNotifications/PushNotificationsBanner/index.tsx

```diff
@@ -95,13 +95,15 @@ export const PushNotificationsBanner = ({ children }: { children: ReactElement }
   const { query } = useRouter()
   const onboard = useOnboard()
 
+  const { getPreferences, getAllPreferences } = useNotificationPreferences()
   const { dismissPushNotificationBanner, isPushNotificationBannerDismissed } = useDismissPushNotificationsBanner()
 
   const isSafeAdded = !!addedSafesOnChain?.[safeAddress]
-  const shouldShowBanner = isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded
+  const isSafeRegistered = getPreferences(safe.chainId, safeAddress)
+  const shouldShowBanner =
+    isNotificationsEnabled && !isPushNotificationBannerDismissed && isSafeAdded && !isSafeRegistered
```

Comment:
Could you rename `isNotificationsEnabled` to something like `isFeatureEnabled`? It's confusing wrt user enabling notifications/allowing push notifications.